### PR TITLE
Feature/ug 1789

### PR DESCRIPTION
**Pull request by AnkurIK**
Please check my PR for ticket no. 1789](https://uplevel-watch.atlassian.net/browse/UG-1789)

Please consider below changes for the review:

- Had refactor the existing code to remove unnecesssary/duplicate lines by creating funcations
- Added click event handlers for the 'Register Now'
- Functionality to set the nearest date based on the date range difference

### CODE REVIEW

File: learn-ik/webinar-learn.js [hunk 1]
```diff
@@ -350,8 +357,10 @@ $(document).ready((function () {
...
       i = v_timezone + ":learn.ik" + cta_lp + ":learn.ik" + getCookie("ik-landingpage-v2");
-      r = "?utm_source=" + $(".utm_source").val() + "&assigned_to=Interview Kickstart&invitee_first_name=" + $(".wr__firstname").val() + "&invitee_last_name=" + $(".wr__lastname").val() + "&invitee_email=" + $(".wr__email").val() + "&answer_1=" + $(".wr__phone").val() + "&event_start_time=" + t + "&event_end_time=" + a + "&utm_medium=" + n + "&salesforce_uuid=" + i;
-      var redirectUrl = "https://learn.interviewkickstart.com/reviews" + r;
+      // r = "?utm_source=" + $(".utm_source").val() + "&webinarType=" + webinarType + "&event=" + eventName + "&invitee_last_name=" + "&event_start_time=" + t + "&invitee_first_name=" + $(".wr__firstname").val() + $(".wr__lastname").val() + "&invitee_email=" + $(".wr__email").val() + "&answer_1=" + $(".wr__phone").val() + "&utm_medium=" + n + "&salesforce_uuid=" + i;
+      r = "?utm_source=" + $(".utm_source").val() + "&webinarType=" + webinarType + "&event=" + eventName + "&event_start_time=" + t + "&utm_medium=" + n + "&salesforce_uuid=" + i;
+      var redirectUrl = "https://ikdev.webflow.io/signup-final-step" + r;
```
Comment (KDholakiya) [inline learn-ik/webinar-learn.js:362]: Why staging link is here? any specific reason for this?

File: learn-ik/webinar-learn.js [hunk 2]
```diff
@@ -665,4 +642,79 @@ $(document).ready((function () {
...
+
+    var expirationTime = new Date(new Date().getTime() + 10 * 365 * 24 * 60 * 60 * 1000);
+    var currentHostname = window.location.hostname;
+    var hostnameParts = currentHostname.split('.');
+    if (hostnameParts.length > 1) {
+      hostnameParts.shift();
+    }
+    var domain = hostnameParts.join('.');
+    setTimeout(() => {
+      console.log('---SET COOKIES---', domain);
```
Comment (KDholakiya) [inline learn-ik/webinar-learn.js:677]: we should remove this.

**Review by KDholakiya — COMMENTED** [review]
will this code be applicable to whole learn site or just these two pages?
1. upsight-sessions/free-data-science-workshop-apr-7
2. career-session-webinar-registration

File: learn-ik/webinar-learn.js (hunk 2 above)
Comment (AnkurIK) [reply learn-ik/webinar-learn.js:677]: I have removed console log

File: learn-ik/webinar-learn.js (hunk 1 above)
Comment (AnkurIK) [reply learn-ik/webinar-learn.js:362]: I have changes Productin URL

**Comment by AnkurIK** [comment]
> will this code be applicable to whole learn site or just these two pages?
> 
> 1. upsight-sessions/free-data-science-workshop-apr-7
> 2. career-session-webinar-registration

---Only Just two page.

**Comment by AnkurIK** [comment]
> will this code be applicable to whole learn site or just these two pages?
> 
> 1. upsight-sessions/free-data-science-workshop-apr-7
> 2. career-session-webinar-registration

This code only for upsight-session page

**Comment by KDholakiya** [comment]
> > will this code be applicable to whole learn site or just these two pages?
> > 
> > 1. upsight-sessions/free-data-science-workshop-apr-7
> > 2. career-session-webinar-registration
> 
> This code only for upsight-session page

If it is only for specific page than shouldn’t we consider to create separate js file for this? it will only add extra weight to other pages where it is not needed.

**Review by KDholakiya — APPROVED** [review]

**Pull request merged**